### PR TITLE
[5.x] Added textProps prop

### PR DESCRIPTION
**Pull request by Peretz30**
***Warning! Breaking change***
Added `textProps` prop that replace existing `textSelectable` and `allowFontScaling`. Now, with this prop you are able to apply any text prop (`selectable`, `textBreakStrategy`, etc.) to text wrapper component.

Example usage:
```
                    <HTML
                        html={
                            newsItem.bodyHtml ||
                            '<div></div>'
                        }
                        imagesMaxWidth={maxWidth}
                        onLinkPress={onLinkPress}
                        tagsStyles={backendStyles.tagsStyles}
                        baseFontStyle={styles.text}
                        textProps={{ textBreakStrategy: "simple" }}
                    />
```

### CODE REVIEW

**Comment by lingtran** [comment]
Awesome, thanks for creating this pull request. I totally understand folks may be swamped at the moment. There have been a number of requests for this fix since Android Q made the text change. For our team we aren't able to simply add the textBreakStrategy prop directly. So we've created a fork with this fix. It would be great if this can be reviewed by a few of the maintainers and merged soon.

**Comment by jjhampton** [comment]
Please review and merge, this is a necessary fix w/ changes that Android Q introduced with text wrapping.

**Comment by kuetsch** [comment]
This would really be helpful! Would love to see the next release incorporate this fix. 
Btw. I think there is one more location where the fix needs to be applied: Its the `textwrapper`-Function in HTMLRenderers.js

**Comment by kamalpandey** [comment]
Waiting for this one to get review and merged.

File: README.md [hunk 1]
```diff
@@ -90,8 +90,7 @@ Prop | Description | Type | Required/Default
 `emSize` | The default value in pixels for `1em` | `number` | `14`
 `ptSize` | The default value in pixels for `1pt` | `number` | `1.3`
 `baseFontStyle` | The default style applied to `<Text>` components | `object` | `{ fontSize: 14 }`
-`allowFontScaling` | Specifies whether fonts should scale to respect Text Size accessibility settings | `boolean` | `true`
-`textSelectable` | Allow all texts to be selected | `boolean` | `false`
+`textProps` | Object with props, that would be applyed to text wrapper. For example `{allowFontScaling: 'false'}` | `object` | `{}`
```
Comment (jsamr) [inline README.md:93]: I would favor renaming this new prop to `textWrapperProps`, let's be explicit!

File: src/HTML.js [hunk 2]
```diff
@@ -64,11 +63,10 @@ export default class HTML extends PureComponent {
         baseFontStyle: { fontSize: 14 },
         tagsStyles: {},
         classesStyles: {},
-        textSelectable: false,
-        allowFontScaling: true
+        textProps: {},
     }
 
-    constructor (props) {
+    constructor(props) {
```
Comment (jsamr) [inline src/HTML.js:69]: It seems like your linter or text editor has decided to reformat the entire file. Could you remove any changes that are not related to the proposed feature?

**Comment by Peretz30** [comment]
@jsamr thanks for your suggestions. Please, check out PR now.

File: src/HTML.js [hunk 3]
```diff
@@ -470,12 +467,11 @@ export default class HTML extends PureComponent {
...
                 convertedCSSStyles
             ]
-            .filter((s) => s !== undefined);
+                .filter((s) => s !== undefined);
 
             const renderersProps = {};
             if (Wrapper === Text) {
-                renderersProps.allowFontScaling = allowFontScaling;
-                renderersProps.selectable = textSelectable;
+                Object.keys(textWrapperProps).map(propName => { renderersProps[propName] = textWrapperProps[propName] });
```
Comment (jsamr) [inline src/HTML.js:474]: I would rather go with:

```javascript
let renderersProps = {};
if (Wrapper === Text) {
    renderersProps = {
        ...textWrapperProps
    }
}
```
which I find easier to read

File: src/HTML.js [hunk 4]
```diff
@@ -305,74 +303,74 @@ export default class HTML extends PureComponent {
...
-                    // If the only child of a node is using the same wrapper, merge them into one
-                    return {
-                        ...parsedNode,
-                        attribs: { ...attribs, ...firstChild.attribs },
-                        data: firstChild.data,
-                        children: [],
-                        tagName,
-                        nodeIndex
-                    };
+            .filter((parsedNode) => parsedNode !== false && parsedNode !== undefined) // remove useless nodes
```
Comment (jsamr) [inline src/HTML.js:306]: Because it is unnecessary formatting, you can reset this hunk to its master state.
To do so, you can do:
```
git checkout --patch src/HTML.js master
```
It will prompt you to select the hunks you want to restore to master's state, and those you want to keep ;-)

File: src/HTML.js [hunk 5]
```diff
@@ -305,74 +303,74 @@ export default class HTML extends PureComponent {
...
+                        ...cssStringToObject(attribs.style || '')
+                    };
 
-                let textChildrenInheritedStyles = {};
-                Object.keys(wrapperStyles).forEach((styleKey) => {
-                    // Extract text-only styles
-                    if (TextOnlyPropTypes.indexOf(styleKey) !== -1) {
-                        textChildrenInheritedStyles[styleKey] = wrapperStyles[styleKey];
-                        delete wrapperStyles[styleKey];
+                    let textChildrenInheritedStyles = {};
```
Comment (jsamr) [inline src/HTML.js:340]: Idem

File: src/HTML.js [hunk 6]
```diff
@@ -305,74 +303,74 @@ export default class HTML extends PureComponent {
...
+                    Object.keys(wrapperStyles).forEach((styleKey) => {
```
Comment (jsamr) [inline src/HTML.js:341]: Idem

File: src/HTML.js [hunk 7]
```diff
@@ -305,74 +303,74 @@ export default class HTML extends PureComponent {
...
+                        // Extract text-only styles
```
Comment (jsamr) [inline src/HTML.js:342]: Idem

File: src/HTML.js [hunk 8]
```diff
@@ -305,74 +303,74 @@ export default class HTML extends PureComponent {
...
-                        // Set (or merge) the inherited text styles extracted from the wrapper for
-                        // each Text child
-                        if (!attribs.style) {
-                            child.attribs.style = cssObjectToString(textChildrenInheritedStyles);
-                        } else {
-                            child.attribs.style = cssObjectToString({
-                                ...textChildrenInheritedStyles,
-                                ...cssStringToObject(child.attribs.style)
-                            });
+                        if (wrapper === 'Text') {
```
Comment (jsamr) [inline src/HTML.js:358]: Idem

File: src/HTML.js [hunk 9]
```diff
@@ -305,74 +303,74 @@ export default class HTML extends PureComponent {
...
+                                    ...cssStringToObject(child.attribs.style)
+                                });
+                            }
                         }
                     }
                 }
-            }
-            return parsedNode;
-        });
+                return parsedNode;
```
Comment (jsamr) [inline src/HTML.js:372]: Idem

**Comment by Peretz30** [comment]
hey @jsamr ! Sorry for late response, I was very busy last week. I cleaned up code formatting. Check this out, please.

File: README.md [hunk 10]
```diff
@@ -90,8 +90,7 @@ Prop | Description | Type | Required/Default
 `emSize` | The default value in pixels for `1em` | `number` | `14`
 `ptSize` | The default value in pixels for `1pt` | `number` | `1.3`
 `baseFontStyle` | The default style applied to `<Text>` components | `object` | `{ fontSize: 14 }`
-`allowFontScaling` | Specifies whether fonts should scale to respect Text Size accessibility settings | `boolean` | `true`
-`textSelectable` | Allow all texts to be selected | `boolean` | `false`
+`textWrapperProps` | Object with props, that would be applyed to text wrapper. For example `{allowFontScaling: 'false'}` | `object` | `{}`
```
Comment (jsamr) [inline README.md:93]: *applied*

File: src/HTML.js [hunk 11]
```diff
@@ -504,7 +500,7 @@ export default class HTML extends PureComponent {
                 remoteErrorView(this.props, this.state) :
                 (
                     <View style={{ flex: 1, alignItems: 'center' }}>
-                        <Text allowFontScaling={allowFontScaling} style={{ fontStyle: 'italic', fontSize: 16 }}>Could not load { this.props.uri }</Text>
+                        <Text { ...textWrapperProps } style={{ fontStyle: 'italic', fontSize: 16 }}>Could not load { this.props.uri }</Text>
```
Comment (jsamr) [inline src/HTML.js:503]: Ibdem

File: src/HTML.js [hunk 12]
```diff
@@ -474,8 +471,7 @@ export default class HTML extends PureComponent {
 
             const renderersProps = {};
```
Comment (07734) [inline src/HTML.js:472]: Const
